### ISSUE TITLE
It is not possible to disable singleAttributePerLine with JSX

### ISSUE THREAD

**Issue by tuyennv-se** [prettier/prettier#16587]
**Prettier 3.3.3**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeACgJwgc03AZwIDoBhAS0zABs4ACA8gLzgF5gBOABgF86AHbHkIF2ACgIwAhjHKTyYAgH5i-OFQQwA+pJl0APvrpcAlHQD0dAIxdedAEYRMAE3UB1cs5gALdnZ8KANZQIuw2fJLYgXCkUvzsAOTYAK5Qzgl8kNRO7FlOJADu3uTwWvY4ETAAnrTswA5SYIF4EKnOpBDZmEh0eZgk5Vpg3lKYWt4QALZwADQOTq6YAEpSzuTJBD0ALACsdDx85gB8IDMgEPyy0ATIoKPYBeijCDcoUtQFUlU3Z-aYjdEYABlKTTAAy5BCyAAZu8CLMQH8AXBgfxGpCcMgYJhkgj4ZNyFicQi4AAPNSYcjTWDvAAq6igo3IhBhcLxGNoAEVkhB4KzqPCzgArAikoEcuDc3lwfmCkAARx58CwF1eICkBAAtCE4K5nKcQNipORqBiOpNJlJkOrqNQDYwoDhaABBGDY8j2ZLK9QQqFIWEChHeGCTahuYrwAhosBwIEvErkABuJSq1sUPxAidxAEk0pogWBKZdnWkgdVaLKEYIIPC3P9+NbBIR1ImZWdIfDMDAsFIcJbK2c0f11Nb7FJ7HA7YPKbAPF5vMgABxcM74RWUOA9vtW-1ss7SexznzIABMZw2cFp49eAblcEmE+cerBUkdyV7cAAYk5LW6MdapC9CAQAOIA)

```sh
--parser babel
```

**Input:**

```jsx
<Progress.Circle size={90} progress={(statistics?.percent_stat || 0) / 100} borderWidth={0} thickness={10} strokeCap={'round'} color={colors.white_bg} style={{ backgroundColor: colors.bg_char_home, borderRadius: 45 }} />
```

**Output:**

```jsx
<Progress.Circle
  size={90}
  progress={(statistics?.percent_stat || 0) / 100}
  borderWidth={0}
  thickness={10}
  strokeCap={"round"}
  color={colors.white_bg}
  style={{ backgroundColor: colors.bg_char_home, borderRadius: 45 }}
/>;

```

**Expected output:**

```jsx
<Progress.Circle size={90} progress={(statistics?.percent_stat || 0) / 100} borderWidth={0} thickness={10} 
  strokeCap={'round'} color={colors.white_bg} style={{ backgroundColor: colors.bg_char_home, borderRadius: 45 }} />
```

**Why?**
When the "singleAttributePerLine" option is not specified or is set to "singleAttributePerLine=false," Prettier still automatically breaks each JSX attribute onto a new line if the total line length exceeds 120 characters (the printWidth value). Currently, based on my research, there is no way to disable this behavior.

I propose changing this option as follows:

Add a new option "forceSingleAttributePerLine" that behaves like the current "singleAttributePerLine=true".
When "forceSingleAttributePerLine" is not set or is set to "forceSingleAttributePerLine=false":
The "singleAttributePerLine=true" option should only apply if the line exceeds the printWidth value.
The "singleAttributePerLine=false" option should prevent line breaks for each attribute.

**Comment by sosukesuzuki** [comment]
I disagree with this proposal.

But if it were to be implemented, instead of adding new option, the value of `singleAttributePerLine` should be `"never" | "auto" | "always"` instead of a boolean value.

**Comment by tuyennv-se** [comment]
@sosukesuzuki 
Yes, the implementation with three values "never" | "auto" | "always" is a great idea.

Regarding your disagreement with the proposal, I still don't fully understand the reasoning. Do you agree with adding the ability to disable the `singleAttributePerLine` feature but disagree with my implementation because it's not optimal? Or do you disagree with adding the ability to disable the `singleAttributePerLine` feature altogether?

**Comment by markau** [comment]
[Overrides ](https://prettier.io/docs/configuration) were the solution here for me, we could keep the `"singleAttributePerLine": true` by default, but exclude that for html files (actually string literals but same result)

```
"singleAttributePerLine": true,
  "overrides": [
    {
      "files": ["templates.ts"],
      "options": {
        "singleAttributePerLine": false
      }
    }
  ]
```

**Comment by waterplea** [comment]
It would be great if `singleAttributePerLine` was instead `maxAttributesPerLine: number`
